<!-- Puts the wheelchair in auto mode and start acml -->
<!-- author: Arturo, Procopio -->
<launch>  
  <node pkg="wheelchair" type="bb_robot" name="bb_robot" output="screen" args="194.199.21.27 1234 1000000" >
    <param name="xScanOffset" value="0.268999993801117"/>
    <param name="yScanOffset" value="0.0"/>
    <param name="startAngle" value="-1.57079632679489661923"/>
    <param name="endAngle" value="1.57079632679489661923"/>
    <param name="rangeMin" value="0.0"/>
    <param name="rangeMax" value="16.0"/>
    <param name="beams" value="181"/>
    <param name="autonomousMode" value="true"/>
    <param name="watchdogDuration" value="0.5"/>
  </node>
  
  <node pkg="amcl" type="amcl" name="amcl" respawn="false" output="screen">
    <remap from="scan" to="base_scan" />
    <remap from="map" to="/map" />

    <!-- Overall filter parameters -->
    <param name="min_particles" value="500"/>
    <param name="max_particles" value="5000"/>
    <param name="kld_err" value="0.05"/>
    <param name="kld_z" value="0.99"/>
    <param name="update_min_d" value="0.2"/>
    <param name="update_min_a" value="0.2"/>
    <param name="resample_interval" value="1"/>
    <param name="transform_tolerance" value="0.2"/>
    <param name="recovery_alpha_slow" value="0.0"/>
    <param name="recovery_alpha_fast" value="0.0"/>
    <param name="gui_publish_rate" value="10.0"/>

    <!-- Laser model parameters -->
    <param name="laser_max_beams" value="30"/>
    <param name="laser_z_hit" value="0.5"/>
    <param name="laser_z_short" value="0.05"/>
    <param name="laser_z_max" value="0.05"/>
    <param name="laser_z_rand" value="0.5"/>
    <param name="laser_sigma_hit" value="0.2"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_model_type" value="likelihood_field"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>

    <!-- Odometery model parameters -->
    <param name="odom_model_type" value="diff"/>
    <param name="odom_alpha1" value="0.2"/>
    <param name="odom_alpha2" value="0.2"/>
    <param name="odom_alpha3" value="0.8"/>
    <param name="odom_alpha4" value="0.2"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="base_frame_id" value="base_link"/>
    <param name="global_frame_id" value="/map"/>

    <param name="initial_pose_x" value="0"/>
    <param name="initial_pose_y" value="0"/>
    <param name="initial_pose_a" value="0"/>

    <param name="use_map_topic" value="true" />
  </node>

</launch>
